internal/app/machined/main.go

```diff
@@ -143,7 +144,19 @@ func main() {
 			Controller: c,
 		}
 
-		e := factory.ListenAndServe(server, factory.Network("unix"), factory.SocketPath(constants.MachineSocketPath))
+		f, e := os.OpenFile(filepath.Join(constants.SystemRunPath, "log", "machined.log"), os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
```

Comment:
you also could potentially use https://github.com/talos-systems/talos/blob/master/internal/app/machined/pkg/system/log/log.go to abstract away a bit this piece